src/controller/java/AndroidICDClient.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ *   Copyright (c) 2024 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+/**
+ *    @file
+ *      Implementation of ICD Client API for Android Platform
+ *
+ */
+
+#include "AndroidICDClient.h"
+
+#include <app/icd/client/ICDClientInfo.h>
+
+chip::app::DefaultICDClientStorage sICDClientStorage;
```

Comment:
Later android tv-app may act as the hub and get the ICD key and offset from cloud and save these icd metadata into ICDClientStorage for later check-in message handling. I think we can further expose the Java API related to https://github.com/project-chip/connectedhomeip/blob/master/src/app/icd/client/ICDClientStorage.h#L38  for ICDClientStorage.

This may be follow-up, not blocking.

Reply:
thank you for telling me. I think this part can be implemented additionally in this file once the exact requirements are determined in tv-app.